### ISSUE TITLE
Boinc android 

### ISSUE THREAD

**Issue by robertduchene** [BOINC/boinc#3357]
**Describe the bug**
A
All Android boinc project downloads fail with computation error.

**Steps To Reproduce**
1. Load any boinc project
2. 

**Expected behavior**

**Screenshots**

**System Information**
 - OS: Android 10 OnePlus 7T
 - BOINC Version: Android beta

**Additional context**

**Comment by adamradocz** [comment]
Which project? Can you try another project?

**Comment by Bronsens** [comment]
I have the same issue on Seti@home Google Pixel 3 XL running Android 10.
![Screenshot_20191110-125604](https://user-images.githubusercontent.com/29926234/68548321-9c0aad80-03b9-11ea-8e95-a67ab58f7444.png)
![Screenshot_20191110-125616](https://user-images.githubusercontent.com/29926234/68548323-a62cac00-03b9-11ea-85c3-a33f278f3045.png)

**Comment by Bronsens** [comment]
This is the report on the WU for the result.
`<core_client_version>7.16.3</core_client_version>
<![CDATA[
<message>
process exited with code 1 (0x1, -255)</message>
<stderr_txt>
CANNOT LINK EXECUTABLE "../../projects/setiathome.berkeley.edu/setiathome_8.00_arm-android-linux-gnu__armv7-vfpv4": "/data/data/edu.berkeley.boinc/client/projects/setiathome.berkeley.edu/setiathome_8.00_arm-android-linux-gnu__armv7-vfpv4" has text relocations (https://android.googlesource.com/platform/bionic/+/master/android-changes-for-ndk-developers.md#Text-Relocations-Enforced-for-API-level-23)

</stderr_txt>
]]>`

**Comment by truboxl** [comment]
I think that's for SETI and the title should be appropriately changed

**Comment by ProDigit** [comment]
Similar behavior on Android 10, Essential phone.
First error: Suspended. Connect your phone to a charger.
It is connected to a charger.
I bypassed it by lowering battery levels from 90% to 50%, (though my battery level is at 100%), and allowing it to crunch on battery too.

Computation Errors on ALL 'Moo! Wrapper' SETI@home, and 'Cruncher ogr tasks, the only ones that are coming in.
It seems to start the task, but after about 5-10 seconds display the Computation error.

**Comment by truboxl** [comment]
As mentioned in https://android.googlesource.com/platform/bionic/+/master/android-changes-for-ndk-developers.md#Text-Relocations-Enforced-for-API-level-23
&
https://github.com/android/ndk/wiki/Changelog-r12-beta2
>The -Wl, --warn-shared-textrel, and -Wl,--fatal-warnings options are now enabled by default. If you have shared text relocations, your app cannot load on Android 6.0 (API level 23) or higher. Text relocations have never been allowed for 64-bit apps.

and as such not a BOINC problem, unless this issue wants BOINC to factor in API levels for proper tasks distribution

**Comment by magicrub** [comment]
Here's a hint about the Android 10 problem and it could be fixed in the realm of BOINC?
System binaries/libraries mapped to execute-only memory
https://developer.android.com/about/versions/10/behavior-changes-all#xom-binaries

Not sure how BOINC calls the individual projects, or if it encapsulates/sandboxes it, but as a developer (un-related to BOINC/SETI) I'm keen to help where I can.

**Comment by sunk818** [comment]
@truboxl I updated my latest Samsung Active Tab2 and BOINC stopped working too. So, it'd be good for BOINC to work with modern Android versions since they can crunch more tasks than older smartphones with Android 4.x. I'm happy to tip $20 EUR (Paypal) for prioritizing a fix and release a BETA build to Android store... @adamradocz

**Comment by robertduchene** [comment]
I wonder how much BOINC Android will affect the projects?  How many people use Android vs PC version?  It will probably come to a head where the Project Creators will force BOINC to update the Android application because fewer and fewer results will be coming in.

**Comment by AenBleidd** [comment]
@sunk818, I don't want to disappoint you, but $20 will not solve or prioritize this issue. And there are several reasons for this.
At first, the complexity of the issue itself. Latest SDKs have very strict policies about background processes. So the solution will be complex. I have some ideas how to fix this but I can't currently do this because of the reason two: time. I have no enough free time now to work on this issue. Neither @adamradocz, I'm afraid. Assuming, that the average hour rate of Software Developer in Europe is around $20, I suppose, that your belief is that this issue could be solved in one hour. But it can't. So please, be patient. I'm working on it, and I'm worrying about Android for BOINC too.
And this $20 you could send to some charity organization, or directly to @adamradocz as a Thanksgiving present 😜

**Comment by adamradocz** [comment]
I still think it's not BOINC related, but SETI. Try run Einstein@home on Android 10 and see what's happening.

**Comment by robertduchene** [comment]
I tried ever Android BOINC project and 100% failed every time.

**Comment by adamradocz** [comment]
That's weird because I tried with WCG OpenZika when there were WUs for android, and it worked for me.

**Comment by Ageless93** [comment]
https://einsteinathome.org/host/12789879/tasks/0/0 shows a host
successfully completing Arecibo tasks on Android 10 (
https://einsteinathome.org/it-it/host/12789879)

**Comment by RichardHaselgrove** [comment]
Using `BOINC client version:7.4.53`

**Comment by RichardHaselgrove** [comment]
Host 12789879 is showing 6 current tasks - the most recent issued on 27 November, using plan class ASIMDPIE.

The owner of host 12789879 posted on the Einstein message boards yesterday (30 November):
> I just upgraded on of my Android devices to Android 10 and it doesn't receive other WUs.

The Einstein server logs confirm this:
```
2019-12-01 05:21:21.6449 [PID=29071]    [version] no app version available: APP#40 (hsgamma_FGRPB1G) PLATFORM#12 (arm-android-linux-gnu) min_version 0
2019-12-01 05:21:21.6449 [PID=29071]    [version] Checking plan class 'NEON'
2019-12-01 05:21:21.6473 [PID=29071]    [version] reading plan classes from file '/BOINC/projects/EinsteinAtHome/plan_class_spec.xml'
2019-12-01 05:21:21.6474 [PID=29071]    [version] CPU [' fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp '] lacks feature ' neon '
2019-12-01 05:21:21.6474 [PID=29071]    [version] Checking plan class 'VFP'
2019-12-01 05:21:21.6474 [PID=29071]    [version] CPU [' fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp '] lacks feature ' vfp '
2019-12-01 05:21:21.6474 [PID=29071]    [version] Checking plan class 'ASIMDPIE'
2019-12-01 05:21:21.6475 [PID=29071]    [version] Couldn't match OS version '4.14.116 (Android 10)' with required regexp 'Android [987N65]|Android 4\.[1-9]|3\.4\.[0-9]'
2019-12-01 05:21:21.6475 [PID=29071]    [version] Checking plan class 'NEONPIE'
2019-12-01 05:21:21.6475 [PID=29071]    [version] CPU [' fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp '] lacks feature ' neon '
2019-12-01 05:21:21.6475 [PID=29071]    [version] no app version available: APP#19 (einsteinbinary_BRP4) PLATFORM#12 (arm-android-linux-gnu) min_version 0
2019-12-01 05:21:21.6481 [PID=29071]    [version] no app version available: APP#49 (einstein_O2AS20-500) PLATFORM#12 (arm-android-linux-gnu) min_version 0
2019-12-01 05:21:21.6551 [PID=29071] [debug]   [HOST#12789879] MSG(high) Invalid global preferences supplied, please check https://einsteinathome.org/community/forum/19 on how to fix that.
2019-12-01 05:21:21.6552 [PID=29071] [debug]   [HOST#12789879] MSG(high) No work sent
2019-12-01 05:21:21.6552 [PID=29071] [debug]   [HOST#12789879] MSG(high) Gamma-ray pulsar binary search #1 on GPUs is not available for your type of computer.
```
I'm afraid this can't be used as an example of the elusive 'working Android 10' case - the tasks visible must have been completed before the version upgrade.

**Comment by sunk818** [comment]
@AenBleidd @adamradocz I'm sorry if I offended anyone. it was more intended as encouragement for all that everyone on the boinc team does. I fully realize $20 doesn't cover anyone's time -- maybe a few beers 🍺🍺🍺 that's all! 😄

**Comment by adamradocz** [comment]
You haven't offended anyone. Actually a system for motivating programmers would be nice. But that's offtopic. :)

**Comment by sunk818** [comment]
strangely, I got einstein@home tasks yesterday although not sure if they completed successfully. rac jumped from 250 to 650. again the Samsung active tab2 has no tasks on known projects that offer Android tasks again. my other 20 Android 4.x smartphones are crunching tasks just fine.

**Comment by RichardHaselgrove** [comment]
Can you post your Einstein account ID so that we can explore the server logs? 'sunk818' doesn't appear to be an active username.

**Comment by sunk818** [comment]
@RichardHaselgrove It was under grcpool.com

I had seti@home but they all ended as computational error. 

whatever is in this new Android version, boinc client no likey.

**Comment by sunk818** [comment]
@RichardHaselgrove I found the work units that are pending on grcpool.

There are many tasks listed as "in progress", but I see no tasks on my Android tablet.

```

TASK ID | WORKUNIT ID | SENT | TIME REPORTED OR DEADLINE | STATUS | RUN TIME | CPU TIME | GRANTED CREDIT | ApplicationAll applications (5)Binary Radio Pulsar Search (Arecibo) (5)
-- | -- | -- | -- | -- | -- | -- | -- | --
p2030.20170425.G45.26+01.80.C.b4s0g0.00000_3711_0 | 428680315 | 30 Nov 2019 2:27:53 UTC | 30 Nov 2019 22:46:31 UTC | Completed, waiting for validation | 73,105 | 54,500 | 0 | Binary Radio Pulsar Search (Arecibo) v1.46 () arm-android-linux-gnu
p2030.20170425.G45.26+01.80.C.b4s0g0.00000_3716_0 | 428680320 | 30 Nov 2019 2:27:53 UTC | 30 Nov 2019 22:36:22 UTC | Completed, waiting for validation | 72,492 | 54,142 | 0 | Binary Radio Pulsar Search (Arecibo) v1.46 () arm-android-linux-gnu
p2030.20170425.G45.26+01.80.C.b4s0g0.00000_3717_1 | 428680321 | 30 Nov 2019 2:27:53 UTC | 30 Nov 2019 22:20:22 UTC | Completed, waiting for validation | 71,541 | 53,411 | 0 | Binary Radio Pulsar Search (Arecibo) v1.46 () arm-android-linux-gnu
p2030.20170425.G45.26+01.80.C.b6s0g0.00000_2697_0 | 428768713 | 30 Nov 2019 2:27:53 UTC | 30 Nov 2019 22:39:16 UTC | Completed, waiting for validation | 72,667 | 54,186 | 0 | Binary Radio Pulsar Search (Arecibo) v1.46 () arm-android-linux-gnu
p2030.20170425.G45.26+01.80.C.b6s0g0.00000_2704_0 | 428768721 | 30 Nov 2019 2:27:53 UTC | 1 Dec 2019 18:31:52 UTC | Completed, waiting for validation | 71,881 | 53,268 | 0 | Binary Radio Pulsar Search (Arecibo) v1.46 () arm-android-linux-gnu


```

**Comment by AenBleidd** [comment]
@sunk818, this means that these tasks were also sent to another PC (it's a normal practice, usually the same task is sent twice to different PC to prevent cheating) and are awaiting for another result to verify that calculation was made correctly

**Comment by RichardHaselgrove** [comment]
Thanks. That enables us to identify your machine as [computer 12796671](https://einsteinathome.org/host/12796671). You have [completed tasks](https://einsteinathome.org/host/12796671/tasks/0/19) which have validated and been granted credit in the list, too.

There is no sign that you requested new work for that device in the [most recent server log](https://einsteinathome.org/host/12796671/log), but that link will show new information after every server contact - you, or others here, may be able to recover useful data that I'm not seeing at this time.

The original question was 

> I updated my latest Samsung Active Tab2 and BOINC stopped working too.

What exactly did you update? The machine is currently showing
```
Operating system: Android 3.18.91-16669390 (Android 9)
BOINC client version: 7.16.3
```
Is it the BOINC, or the Android version, which is new?

**Comment by sunk818** [comment]
@RichardHaselgrove The Android version is upgraded. I wish I knew how to downgrade it so I can validate this is the specific cause.

As for BOINC client version, I typically use the BETA version of BOINC Android (because I wanted the rename feature). I tried the latest BETA as well as the non-BETA from the Play Store. Neither version are getting tasks (arm-android) from any the projects I've tried (Asteroids@Home, Einstein@Home, Universal@Home, SETI@Home, World Community Grid).

**Comment by RichardHaselgrove** [comment]
I'm not an Android specialist, so I'll leave that to @AenBleidd. But I do have some experience in digging out diagnostics from BOINC logs and websites. Einstein is a useful project in this respect, because it allows outsiders (limited) access to its server logs. So far, your machine hasn't updated beyond the rather limited contact at 11:44 this morning: it seemed BOINC was working then, but didn't have anything useful to say.

According to the Einstein server, it thinks it sent you 320 new tasks between 02:00 and 02:55 UTC today - but you say no tasks are visible on your device. It might be helpful for you to try and find the local logs on your machine relating to those transactions, but I gather that's difficult on Android.

**Comment by sunk818** [comment]
@RichardHaselgrove I am not familiar with the gui debug options. 

There's a value of 2 for debug. Is that the right debug level?

And then there's many different checkboxes. Seems like the log is limited in depth and what I mostly see is HTTP GET/POST, truncated XML, and a few things about requesting tasks. I'm not clear what is useful.

**Comment by RichardHaselgrove** [comment]
Initially, just the raw log itself, without any of the optional checkboxes. The HTTP GET/POST and the XML suggest you have some debug options selected - at this stage, they're just noise, and waste lines in the log. The BOINC log may be different from other ones you've worked with - the checkboxes control what is recorded in real time, they aren't filters on an otherwise complete log.

The default options are <file_xfer>, <sched_ops>, and < task >. I'd add <sched_op_debug>, but no others.

Then, we probably have to wait until BOINC requests work again - with extra logging, the lines for 02:00 - 03:00 this morning have probably scrolled off the display already. I'm in UTC time zone, so off to bed soon. If you manage to register any 'Sending scheduler request' events overnight, please copy them and the following lines here - I can look at them in the morning.

**Comment by RichardHaselgrove** [comment]
I'm no longer sure this is just an Android 9 issue. @sunk818's machine at Einstein is described as

`Android 3.18.91-16669390 (Android 9)`

The Black Friday offers were still active yesterday, so I went out and bought a Samsung Tab 'A' - [Einstein host 12797337](https://einsteinathome.org/host/12797337) - it was frustrating not being able to see these problems for myself. My machine is described as

`Android 4.4.111-16611318 (Android 9)`

and it's completed 12 Einstein tasks overnight. Mind you, that's with BOINC v7.4.53 (all I could find on day one) - I'll report how I get on once I've found the download for v7.16.3

**Comment by sunk818** [comment]
@RichardHaselgrove I'll try to side load an earlier version. I was using grcpool but maybe it'll be easier to use my own ID on the various Android friendly projects.

I tried to downgrade my Samsung Active Tab 2 (different from what I bought) from Android 9 back to 7.1 but there's some sort of Samsung security to prevent loading of any new firmware. Maybe it is an Enterprise feature.

**Comment by RichardHaselgrove** [comment]
I think I have some further evidence on this, pointing to a BOINC v7.16.3 problem.

New tablet has been running since Thursday. Initial (over-large) download of tasks has been completed, and new work fetched from Einstein. Started work on tablet today - timeline:

12:50-ish: Opened tablet, found large Android update waiting for installation.
12:53: Android update complete, tablet restarted.
12:55: BOINC v7.4.53 downloaded new work.
12:57: I downloaded BOINC v7.16.3 APK.
13:00 and 13:11: BOINC v7.16.3 requested new work, got none.
13:19: BOINC both reported and requested new work. Server log:
```
2019-12-08 13:19:39.4420 [PID=25416]    [handle] [HOST#12797337] [RESULT#903314061] [WU#428579429] got result (DB: server_state=4 outcome=0 client_state=0 validate_state=0 delete_state=0)
2019-12-08 13:19:39.4421 [PID=25416]    [handle] cpu time 23714.360000 credit/sec 0.004481, claimed credit 106.275031
2019-12-08 13:19:39.4421 [PID=25416]    [handle] [RESULT#903314061] [WU#428579429]: setting outcome SUCCESS
2019-12-08 13:19:39.4491 [PID=25416]    [send] effective_ncpus 4 max_jobs_on_host_cpu 999999 max_jobs_on_host 999999
2019-12-08 13:19:39.4492 [PID=25416]    [send] effective_ngpus 0 max_jobs_on_host_gpu 999999
2019-12-08 13:19:39.4492 [PID=25416]    [send] Not using matchmaker scheduling; Not using EDF sim
2019-12-08 13:19:39.4492 [PID=25416]    [send] CPU: req 0.00 sec, 0.00 instances; est delay 0.00
```
Einstein server has consistently reported `CPU: req 0.00 sec` ever since the BOINC upgrade. I don't think I have access to the sched_request file unless I wireshark it, but it feels as if it's missing something.

**Comment by sunk818** [comment]
Here's a sample of mine.

Android version 9, Kernel 3.18.91-16669390 (Sep 5 2019)
BOINC 7.4.43 (tried an old version to see if I got different results from earlier).

```
Sat Dec 07 22:53:13 PST 2019|Einstein@Home|[work_fetch] share 0.000 project is backed off┬á (resource backoff: 6840.62, inc 38400.00)
Sat Dec 07 22:53:13 PST 2019|Einstein@Home|[work_fetch] share 1.000
Sat Dec 07 22:53:13 PST 2019|Einstein@Home|[work_fetch] REC 172.539 prio -25.307 can request work
Sat Dec 07 22:52:13 PST 2019|Einstein@Home|[work_fetch] share 0.000 project is backed off┬á (resource backoff: 6900.80, inc 38400.00)
Sat Dec 07 22:52:13 PST 2019|Einstein@Home|[work_fetch] share 0.500
Sat Dec 07 22:52:13 PST 2019|Einstein@Home|[work_fetch] REC 172.420 prio -26.310 can request work
Sat Dec 07 22:51:13 PST 2019|Einstein@Home|[work_fetch] share 0.000 project is backed off┬á (resource backoff: 6960.91, inc 38400.00)
Sat Dec 07 22:51:13 PST 2019|Einstein@Home|[work_fetch] share 0.500
Sat Dec 07 22:51:13 PST 2019|Einstein@Home|[work_fetch] REC 172.301 prio -26.314 can request work
Sat Dec 07 22:50:13 PST 2019|Einstein@Home|[work_fetch] share 0.000 project is backed off┬á (resource backoff: 7021.01, inc 38400.00)
Sat Dec 07 22:50:13 PST 2019|Einstein@Home|[work_fetch] share 0.500
Sat Dec 07 22:50:13 PST 2019|Einstein@Home|[work_fetch] REC 172.182 prio -26.317 can request work
Sat Dec 07 22:49:13 PST 2019|Einstein@Home|[work_fetch] share 0.000 project is backed off┬á (resource backoff: 7081.12, inc 38400.00)
Sat Dec 07 22:49:13 PST 2019|Einstein@Home|[work_fetch] share 0.500
Sat Dec 07 22:49:13 PST 2019|Einstein@Home|[work_fetch] REC 172.063 prio -26.320 can request work
Sat Dec 07 22:48:13 PST 2019|Einstein@Home|[work_fetch] share 0.000 project is backed off┬á (resource backoff: 7141.20, inc 38400.00)
Sat Dec 07 22:48:13 PST 2019|Einstein@Home|[work_fetch] share 0.500
Sat Dec 07 22:48:13 PST 2019|Einstein@Home|[work_fetch] REC 171.944 prio -26.324 can request work
Sat Dec 07 22:47:13 PST 2019|Einstein@Home|[work_fetch] share 0.000 project is backed off┬á (resource backoff: 7201.30, inc 38400.00)
Sat Dec 07 22:47:13 PST 2019|Einstein@Home|[work_fetch] share 0.500
Sat Dec 07 22:47:13 PST 2019|Einstein@Home|[work_fetch] REC 171.825 prio -26.327 can request work
Sat Dec 07 22:46:13 PST 2019|Einstein@Home|[work_fetch] share 0.000 project is backed off┬á (resource backoff: 7261.45, inc 38400.00)
```

**Comment by magicrub** [comment]
As a reminder to this thread, the big offender seems to be Android 10 (skd29) related. Chasing down Android 6,7,8,9 non-problems may not be productive. unless we can see a working version not-working comparison in the logs that can tell us something about the main problem (whatever that problem is, project/boinc or 32/64bit ect ect)

**Comment by RichardHaselgrove** [comment]
Created new issue at #3397, because my problem today was a 'BOINC for Android' problem, not a pure Android problem.

**Comment by magicrub** [comment]
Perhaps we're tracking this down the wrong way. The developers who have created Android related PRs in the last year (merged or closed) should have some insight and/or facing similar problems if they have an Android 10 device.
https://github.com/BOINC/boinc/pulls?q=is%3Apr+android+is%3Aclosed

@adamradocz is the only one from that list who has said anything on this thread. (Thank you!) but also special mention to @RichardHaselgrove You guys seem to be the most active with Android development so maybe we're already talking to the right person. AenBledidd seems to be the next best candidate. I will now shamelessly tag others from the PR list on this issue in case they have any insight. At the very least they should be able to compile+run the simulator on Android 10 (vs 6,7,8,9) and see a fail vs working behavior to sanity check this issue.
@AenBleidd  @okan35 @davidpanderson @nilsjh @brevilo

**Comment by AenBleidd** [comment]
So this issue is happened because since BOINC Android 7.16.3 target SDK version was increased to 28. In order to support newer devices (with Android 10) and latest version of BOINC Android, applications should be rebuilt with newer toolkit (28).
I'm willing to close this ticket as not related to BOINC Android itself.
@davidpanderson, is it possible to rebuilt SETI Android binaries on your side?

**Comment by AenBleidd** [comment]
Closing this as a duplicate of #3290
